### PR TITLE
Add runtime configuration property `quarkus.datasource.active`

### LINKED ISSUE #37265: Support for de-activating a datasource at runtime (application startup)
### Description

As discussed on [Quarkus dev](https://groups.google.com/g/quarkus-dev/c/AnJLtYwgQxs/m/pYr1njhCAgAJ), I would like to have a way to disable datasources at runtime. Currently, this is not possible, I have tried defining different datasources for each flavor, but this still means that the datasource is created for the unused flavours, so I guess it still has some impact. I know that I can remove the unused datasource from the health checks, but I'm assuming (at least that's what I saw in my testing) that the beans are still created.

### Implementation ideas

Essentially, what I'm asking for is something similar to https://quarkus.io/version/main/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.active but for the datasource itself.

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #37265 

~~Creating as draft, because this is based on #38106 , which should be merged first.~~ => Done

I tried to explain the feature in the documentation: see changes to `docs/src/main/asciidoc/datasource.adoc` and `docs/src/main/asciidoc/hibernate-orm.adoc`. Please let me know if something is not clear.

Best reviewed commit-by-commit: I split the implementation and tests. Tests are numerous, part because of the widespread impact of disabling a datasource (on flyway, metrics, health, ...), part because of the duplication of SQL client (needs one test per DB vendor).

### CODE REVIEW

File: docs/src/main/asciidoc/datasource.adoc [hunk 1]
```diff
@@ -405,6 +405,99 @@ AgroalDataSource usersDataSource;
...
+will make only the PostgreSQL datasource available,
+and setting `quarkus.datasource."oracle".active=true` at runtime
+will make only the Oracle datasource available.
+
+[TIP]
+====
+xref:config-reference.adoc#custom-profiles[Custom configuration profiles] can help simplify such a setup.
+By appending the following profile-specific configuration to the one above,
+you can select a persistence unit/datasource at runtime simply by passing
+xref:config-reference.adoc#multiple-profiles[setting configuration property `quarkus.profile`]:
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource.adoc:454]: > simply by passing setting configuration property `quarkus.profile`

sounds weird?

**Review by gsmet — COMMENTED** [review]
Looks good. I think I spotted a small typo in the doc, could you have a look?

File: docs/src/main/asciidoc/datasource.adoc (hunk 1 above)
Comment (yrodiere) [reply docs/src/main/asciidoc/datasource.adoc:454]: Right. I rephrased to:

> simply by setting `quarkus.profile`

(and I did the same for the other occurrence)

**Review by yrodiere — COMMENTED** [review]
Thanks! I addressed your comment and rebased on `main`.

**Review by gsmet — APPROVED** [review]

**Pull request merged**